packages/react-google-maps-api/src/components/addons/MarkerClusterer.tsx

```diff
@@ -128,51 +131,72 @@ export class MarkerClustererComponent extends PureComponent<
     markerClusterer: null
   }
 
-  componentDidMount = () => {
+  // eslint-disable-next-line @getify/proper-arrows/this, @getify/proper-arrows/name
+  setMarkerClustererCallback = (): void => {
```

Comment:
We should avoid using class methods as arrow function. 
See: https://medium.com/@rohans16996/dark-side-of-the-fat-arrow-function-8d2a477f743b

TL;DR - when using arrow function as class properties the function get's duplicated on each component instance. Usually this is negligible, but in our lib you might be rendering out thousands of components so this starts to add up

Reply:
>>>
It`s generally advised to keep *bound* functions out of React Components.
In this case, every single time the component instance is created, a new bound function will be created. which in turn slows down React rendering! Hence, AVOID!

Nothing says about arrow functions in React components, only bound.

Reply:
React Components lifecycle methods does not required to be bound, but all other methods are required. Since React async, we are generally avoiding constructors, so the only one way is to keep all instance methods as arrow functions, if we use this in function scope. Otherwise we can just extract that function outside the module to `./src/utils/` or elsewhere.

Reply:
Why should we avoid constructors? The only issue about constructor function is that it shouldn't contain side effects. Other than that it's fine.
As you say life cycle methods are automatically bound, but any other function (using this) has to be bound. Using the arrow function syntax compiles to defining the function per instance as the article says

Reply:
Why should we avoid constructors? The only issue about constructor function is that it shouldn't contain side effects. Other than that it's fine.
As you say life cycle methods are automatically bound, but any other function (using this) has to be bound. Using the arrow function syntax compiles to defining the function per instance as the article says

Reply:
but bound functions also goes into the object, so there is no gain)

Reply:
everything in constructor could be called multiple times, so better to do not run constructor at all in Components. Soon we will migrate to hooks, and there is no constructors.

in react component all functions goes into an object, except if you are extending some other class, which extends Component or PureComponent.

all lifecycle methods are bound by react, all custom function methods should be bound in constructor. bounding a function returns a copy of the function.

Arrow class methods don't have to be bound in babel environment cos we have a proposal babel plugin to handle that case.

Reply:
You are absolutely... right! I just read this: https://github.com/facebook/react/issues/9851
Ok so that's ok then.